ohttp/build.rs

```diff
@@ -133,59 +133,107 @@ mod nss {
         assert!(status.success(), "NSS build failed");
     }
 
-    fn dynamic_link() {
-        let libs = if env::consts::OS == "windows" {
-            &["nssutil3.dll", "nss3.dll"]
+    fn nspr_libs() -> Vec<&'static str> {
+        if env::consts::OS == "windows" {
+            vec!["libplds4", "libplc4", "libnspr4"]
         } else {
-            &["nssutil3", "nss3"]
-        };
-        dynamic_link_both(libs);
+            vec!["plds4", "plc4", "nspr4"]
+        }
     }
 
-    fn dynamic_link_both(extra_libs: &[&str]) {
-        let nspr_libs = if env::consts::OS == "windows" {
-            &["libplds4", "libplc4", "libnspr4"]
+    fn dynamic_link() {
+        let mut libs = if env::consts::OS == "windows" {
+            vec!["nssutil3.dll", "nss3.dll"]
         } else {
-            &["plds4", "plc4", "nspr4"]
+            vec!["nssutil3", "nss3"]
         };
-        for lib in nspr_libs.iter().chain(extra_libs) {
+
+        libs.append(&mut nspr_libs());
+
+        for lib in &libs {
             println!("cargo:rustc-link-lib=dylib={}", lib);
         }
     }
 
-    fn static_link() {
+    fn static_softoken_libs(target_arch: &str) -> Vec<&'static str> {
+        let mut static_libs = vec!["pk11wrap_static", "softokn_static", "freebl_static"];
+
+        // Hardware-specific acceleration libraries.
+        //   https://searchfox.org/nss/rev/a9928ea43c27da7a25476c07a60fe4779b44f81a/lib/freebl/freebl.gyp#508-542
+        match target_arch {
+            "x86" | "x86_64" => {
+                // NOTE: The intel-gcm-* libraries are already automatically
+                //       included in freebl_static as source files.
+
+                static_libs.push("gcm-aes-x86_c_lib");
+                static_libs.push("sha-x86_c_lib");
```

Comment:
The rules for inclusion of these libraries are somewhat more complex than this.  Would it be possible to scan for the existence of the relevant files instead?  Scanning for files would only require that you have a listing of the different names that might be present here.